src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -623,48 +639,23 @@ namespace SystemAvailabilityManager {
                     ErrorsFound = true;
                 }
 
-                {
-                    auto const SELECT_CASE_var(UtilityRoutines::MakeUPPERCase(cAlphaArgs(4)));
-                    if (SELECT_CASE_var == "STAYOFF") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = StayOff;
-                    } else if (SELECT_CASE_var == "CYCLEONANY") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAny;
-                    } else if (SELECT_CASE_var == "CYCLEONCONTROLZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnControlZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYZONEFANSONLY") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = ZoneFansOnly;
-                    } else if (SELECT_CASE_var == "CYCLEONANYCOOLINGORHEATINGZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyCoolingOrHeatingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYCOOLINGZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyCoolingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYHEATINGZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyHeatingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYHEATINGZONEFANSONLY") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyHeatingZoneFansOnly;
-                    } else {
-                        ShowSevereError(state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + cAlphaArgs(1) + "\", invalid");
-                        ShowSevereError(state, std::string{RoutineName} + "incorrect value: " + cAlphaFieldNames(4) + "=\"" + cAlphaArgs(4) + "\".");
-                        ErrorsFound = true;
-                    }
+                state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).nightCycleControlType = static_cast<NightCycleControlType>(
+                    getEnumerationValue(NightCycleControlTypeNamesUC, UtilityRoutines::MakeUPPERCase(cAlphaArgs(4))));
+
+                if (state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).nightCycleControlType == NightCycleControlType::Invalid) {
```

Comment:
These validity checks are done by the epJSON processor now. No need to do them here. May want to replace this with an `assert( ... != NightCycleControlType::Invalid);`